### PR TITLE
♻️ force failure in pre-reqs to avoid having to manually fail in…

### PR DESCRIPTION
**Pull request by eliascodes**
fixes TwinePlatform/twine-api#132

### Changes
- Add `requireTHING` pre-requisites that fail if `isTHING` pre-req returns false

### Testing Requirements
N/A just a refactor

### Release Requirements
N/A

### Manual Deployment
- API

### CODE REVIEW

File: api/src/api/v1/prerequisites/require_child_organisation.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * "organisations_details-child" route pre-requisite
+ *
+ * Determines if the authenticated user is trying to access an organisation
+ * which is actually a "child" entity.
+ *
+ * Conditions under which this is true for organisation X:
+ * - User is an admin for a funding body which owns at least
```
Comment (astroash) [inline api/src/api/v1/prerequisites/require_child_organisation.ts]: "funding body" concept no longer exists. I think we shouldn't include this in our descriptions anymore

File: api/src/api/v1/prerequisites/require_child_user.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * "user_details-child" route pre-requisite
+ *
+ * Determines if the authenticated user is trying to access an user
+ * which is actually a "child" entity.
+ *
+ * Conditions under which this is true for user X attemptying to access:
+ * user Y
+ * - User X is an admin for a community business for which user Y is a VISITOR
```
Comment (astroash) [inline api/src/api/v1/prerequisites/require_child_user.ts]: Does this return true for siblings too?
ie 

- User X is an volunteer admin for a community business for which user Y is a volunteer admin

Comment (eliascodes) [reply api/src/api/v1/prerequisites/require_child_user.ts]: no, only CB admins, check the is_child_user pre-req

Comment (astroash) [reply api/src/api/v1/prerequisites/require_child_user.ts]: Can this be specified in the description

**Pull request merged**